doc/_index.md

```diff
@@ -1,13 +1,85 @@
+([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph)).  There is
+typically one root node for each kernel revision with lots of child nodes
+containing all the test data that relates to it.
+
+Each node object also has a state which can be used when orchestrating the
+pipeline.  For example, a node will be in the Running state while awaiting some
+results.  There's also a result value, to tell whether the job passed or
+failed.  Finally there's a list of artifacts with URLs to know where to find
+all the related files (binaries, logs, generated results etc.).
+
+Here's a publicly available
+[graph](https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/public/dashboards/5b52c3c6-81bb-4658-a5f3-8ebd2e980436#)
+showing the number of nodes added to the database every day:
+
+<div style="padding: 0 0 20px 0"><iframe style="background: #FFFFFF;border: none;border-radius: 2px;box-shadow: 0 2px 10px 0 rgba(70, 76, 79, .2);" width="640" height="480" src="https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/embed/charts?id=64c9fa16-ba81-40e9-8cd7-cd56fb88fbf3&maxDataAge=3600&theme=light&autoRefresh=true"></iframe></div>
+
+> **Note:** The API doesn't manage storage, the only requirement is to provide
+> publicly-available HTTP(s) URLs for each artifact.
+
+### Publisher / Subscriber Interface
+
+Every time some data changes in the database, basically every time a node is
+has been added or updated, an event is sent on the Publisher / Subscriber
+interface (Pub/Sub).  For example, when a new kernel revision is found and a
+new node is created for it, an event will be sent to tell subscribers about it
+with something like "A checkout node has been created for kernel revision
+v6.2-rc4".  The actual event is a [CloudEvents](https://cloudevents.io) object
+with some JSON data containing a subset of the corresponding node database
+entry.
+
+Any client code can subscribe to receive events with an API token and implement
+features based on how to handle these events.  The API generates events
+automatically whenever nodes are changing but clients may also use the
+interface to publish their own events and coordinate other parts of the
+pipeline too.
+
+### User Ownership
+
+Interacting with the API requires a token which is associated with a particular
+user.  Whenever a user sends some data such as a new node, it is owned by that
+user.  While all nodes are publicly readable, only the owner of the node can
+update it.  Users can also belong to groups to share data with other users.
+
+While the main KernelCI jobs will be creating nodes with users from a
+particular `kernelci.org` group, all the other users can create their own data
+which will coexist in the database.  Then your own nodes can have parents
+created by users.  For example, you may submit test results that relate to a
+kernel build provided by KernelCI.
+
+## Pipeline
+
+The Pipeline is all the client-side services which are running the actual jobs.
```

Comment:
I might be missing some context (same as in L27) on what a "job" is.

I remember there was an attempt to make [test terms glossary](https://elinux.org/Test_Glossary). There's a "Run (noun)" term for what I believe a "job" in KCI API context might be - is it correct? Also, I'm not sure how widely this glossary is used.

Reply:
It's true that it's not very well defined yet.  I'll rewrite this with some paraphrasing to avoid just using the word "job" as a wildcard.

Basically it's what you get when you run `kci job submit` etc. but it's very implicit for an overview documentation page.

Reply:
I think a "run" is a close match but it's a bit unclear in my opinion.  Also "job" means something specific in Kubernetes so it's probably better to avoid using it here to avoid confusion.  And yes while this glossary is very nicely put together I don't think it's an authoritative standard across the whole ecosystem so we can't just assume that everyone will follow the same definitions.

Reply:
Alright I've rephrased these parts and now there's no mention of "job" on this page.  I've replaced it with things like "pipeline steps", basically it's about what the pipeline effectively does.  Please let me know if this is OK.